core/network/warp/sync.cpp

```diff
@@ -71,18 +71,24 @@ namespace kagome::network {
       auto authorities =
           authority_manager_->authorities(block_tree_->getLastFinalized(), true)
               .value();
-      if (not grandpa_->verifyJustification(fragment.justification,
-                                            *authorities)) {
-        return;
-      }
-      Op op{
-          block_info,
-          fragment.header,
+      grandpa_->verifyJustification(
           fragment.justification,
           *authorities,
-      };
-      db_->put(storage::kWarpSyncOp, scale::encode(op).value()).value();
-      applyInner(op);
+          [wptr{weak_from_this()},
+           op{Op{
```

Comment:
Must remove callback, or rewrite like `ok = await verifyJustification`,
otherwise next iteration `verifyJustification` won't succeed without `applyInner` updated authorities/state/info